mesonbuild/interpreter.py

```diff
@@ -2964,10 +2964,15 @@ def evaluate_subproject_info(self, path_from_source_root, subproject_dirname):
         depth = 0
         subproj_name = ''
         segs = path_from_source_root.split(os.path.sep)
-        while segs and segs[0] == subproject_dirname:
-            depth += 1
-            subproj_name = segs[1]
-            segs = segs[2:]
+        segs_spd = subproject_dirname.split(os.path.sep)
```

Comment:
Please use `pathlib.PurePath` for all path manipulation. This will break if you use 'foo/bar' on Windows.